File: robot_examples/go/rest_api/swagger/BoardInfo.go
package swagger

import (
)

type BoardInfo struct {
    MaxBoardX  float32  `json:"maxBoardX,omitempty"`
    MaxBoardY  float32  `json:"maxBoardY,omitempty"`
    StreamDelay  float32  `json:"streamDelay,omitempty"`
    TurnDeltaTime  float32  `json:"turnDeltaTime,omitempty"`
    NetworkLatency  float32  `json:"networkLatency,omitempty"`
    StartTime  float32  `json:"startTime,omitempty"`
    EndTime  float32  `json:"endTime,omitempty"`
    Events  []Event  `json:"events,omitempty"`
    
}

File: robot_examples/haskell-servant/rest_api/lib/Model/BoardInfo.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}

module Model.BoardInfo
    ( BoardInfo (..)
    ) where

import Data.Aeson
import GHC.Generics
import Test.QuickCheck
import Model.Event


data BoardInfo = BoardInfo
    { maxBoardX :: Float
    , maxBoardY :: Float
    , streamDelay :: Float
    , turnDeltaTime :: Float
    , networkLatency :: Float
    , startTime :: Float
    , endTime :: Float
    , events :: [Event]
    } deriving (Show, Eq, Generic)

instance FromJSON BoardInfo
instance ToJSON BoardInfo
instance Arbitrary BoardInfo where
    arbitrary = BoardInfo <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
